### PR TITLE
[Merged by Bors] - TO-3390 fix clippy link

### PR DESCRIPTION
**Pull request by rustonaut**
**References:**

- [TO-3390]

[TO-3390]: https://xainag.atlassian.net/browse/TO-3390?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine_core/bindings/src/lib.rs [hunk 1]
```diff
@@ -505,7 +505,7 @@ impl XaynDiscoveryEngineAsyncFfi {
 
     /// Disposes the engine.
     pub async fn dispose(engine: Box<SharedEngine>) {
-        engine.as_ref().as_ref().lock().await;
+        drop(engine);
```
Comment (rustonaut) [inline discovery_engine_core/bindings/src/lib.rs:508]: Any idea why we acquired and directly dropped the lock before dropping the lock with the engine?

**Review by janpetschexain — APPROVED** [review]

**Comment by rustonaut** [comment]
bors merge

**Comment by rustonaut** [comment]
bors merge

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**